facilitator/src/kubernetes.rs

```diff
@@ -0,0 +1,138 @@
+use anyhow::{anyhow, Result};
+
+use kube::{
+    api::{Api, ListParams},
+    Client,
+};
+
+use k8s_openapi::{api::core::v1::Secret, Metadata};
+
+use tokio::runtime::Runtime;
+
+/// Definition of a namespaced Kubernetes API implementation
+#[derive(Debug)]
+pub struct Kubernetes {
+    namespace: String,
+}
+
+/// Methods for retrieving time from a value
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+struct Time(chrono::DateTime<chrono::Utc>);
+trait WithTime {
+    fn get_time(&self) -> Time;
+}
+
+/// Implementation of WithTime for Kubernetes Secret. Will panic if timestamp
+/// does not exist.
+impl WithTime for Secret {
```

Comment:
I think this is fine, but I think the more idiomatic Rusty thing would be to implement [`std::convert::From<Secret>`](https://doc.rust-lang.org/std/convert/trait.From.html) on `Time`.